scripts/pi-hole/php/password.php

```diff
@@ -68,9 +68,15 @@ function verifyPassword($pwhash, $use_api = false)
                 $_SESSION['auth'] = true;
 
                 // Login successful, redirect the user to the original requested page
-                if ($_SERVER['REQUEST_METHOD'] === 'POST' && $_SERVER['SCRIPT_NAME'] === '/admin/login.php') {
-                    header('Location: '.$redirect_url);
-                    exit;
+                if ($_SERVER['REQUEST_METHOD'] === 'POST') {
+                    // suffix match (str_ends_with()) for /login.php
+                    $script_name = $_SERVER['SCRIPT_NAME'];
+                    if (strlen($script_name) > 10
+                        ? substr_compare($script_name, '/login.php', -10)
+                        : $script_name === '/login.php') {
+                        header('Location: '.$redirect_url);
+                        exit;
```

Comment:
I think this code will work for any `$_SERVER['SCRIPT_NAME']` length:

```suggestion
                if ($_SERVER['REQUEST_METHOD'] === 'POST' && substr_compare($_SERVER['SCRIPT_NAME'], '/login.php', -10) === 0) {
                    header('Location: '.$redirect_url);
                    exit;
```

Reply:
OK.
I think my suggestion might generate warnings on PHP versions older than 7.2.18 (I'm unable to test it right now).

If that's the case, what do you think about this?

```suggestion
                if (
                    $_SERVER['REQUEST_METHOD'] === 'POST' &&
                    strlen($_SERVER['SCRIPT_NAME']) >= 10 &&
                    substr_compare($_SERVER['SCRIPT_NAME'], '/login.php', -10) === 0
                ) {
                    header('Location: '.$redirect_url);
                    exit;
```

Reply:
> OK. I think my suggestion might generate warnings on PHP versions older than 7.2.18 (I'm unable to test it right now).
> 
> If that's the case, what do you think about this?
[snip]

What I think: I am confused.  Are you reviewing my code, which should work portably?  Do you know why my code might not work portably?  Could you please give me feedback on that?  If you are suggesting changes, would you please help me to understand why the changes you are suggesting are better, and what issues you are trying to solve by suggesting changes?  Is it code syntax?  code formatting?  Something else?

I am unable to evaluate your suggestions without knowing more.  There are some potential limitations in [substr_compare()](https://www.php.net/manual/en/function.substr-compare.php) in older versions of PHP, which might generate a warning for certain string lengths of `haystack`, the first argument.  That is why my code uses a ternary expression to handle those cases without issuing a warning.   I have not tested older PHP versions with `needle` longer than `haystack`, but my code is paranoid to avoid doing so.  Maybe the paranoia is unnecessary?  Even if it is too paranoid, does this need to be optimized?  

Note: there is a bug in the code which I will fix momentarily: `substr_compare() == 0` must be the test for success.

Reply:
To hopefully shortcut some back-and-forth: PHP is not my personal language of choice.  If you tell me to "change the code to X", then I will change the code to X.

Reply:
I'm not trying to impose my code or trying to "teach" you anything.

I suggested this change because the use of the ternary operator inside an expression for `if` is confusing to understand and can lead to "hard to find bugs".

My suggestion tests the haystack length before calling `substr_compare()`. 
If the length test fails, `substr_compare()` will never be executed and there will be no warnings.

You can use whatever you like. I just suggested a different one for readability (and it adds the missing `=== 0`).

Reply:
> I suggested this change because the use of the ternary operator inside an expression for if is confusing to understand and can lead to "hard to find bugs".

That may be more true for nested ternary operations (`a < b ? b < c ? c  : b : a`).  However, wrapping long lines of code into multiple lines is very similar in "complexity" when using `&&` or `?:`.

Since you're the one doing the review, I'll implement your suggestion (including inefficiently performing 3x the lookup of `$_SERVER['SCRIPT_NAME']`)